heatpump.yaml

```diff
@@ -2073,52 +2091,157 @@ switch:
+    optimistic : true
+    lambda: "return (id(unmasked_value_register_0) & 0x1) == 0x1;"
+    on_turn_on:
+      #- logger.log: "Switch Turned On!"
+      - lambda: |-
+          uint16_t checked_bit = 0x1;
+          uint16_t new_value = id(unmasked_value_register_0);  // The original unmasked value
+          new_value &= ~checked_bit;         //Clear and set
+          new_value += checked_bit;
+          if ((new_value) != id(unmasked_value_register_0)) {
+            ESP_LOGI("unmasked_value_register_0", "Set option to on power_air_conditioner_zone_1 0x%x -> 0x%x", id(unmasked_value_register_0), new_value);
+            id(unmasked_value_register_0) = new_value;
+            esphome::modbus_controller::ModbusCommandItem set_payload_command = esphome::modbus_controller::ModbusCommandItem::create_write_single_command(${devicename}, 0x0, new_value);
+            ${devicename}->queue_command(set_payload_command);
+          }
+    on_turn_off:
+      #- logger.log: "Switch Turned Off!"    
+      - lambda: |-
+          uint16_t checked_bit = 0x1;
+          uint16_t new_value = id(unmasked_value_register_0);  // The original unmasked value
+          new_value &= ~checked_bit;         //Clear and set
+          if ((new_value) != id(unmasked_value_register_0)) {
+            ESP_LOGI("unmasked_value_register_0", "Set option to off power_air_conditioner_zone_1 0x%x -> 0x%x", id(unmasked_value_register_0), new_value);
+            id(unmasked_value_register_0) = new_value;
+            esphome::modbus_controller::ModbusCommandItem set_payload_command = esphome::modbus_controller::ModbusCommandItem::create_write_single_command(${devicename}, 0x0, new_value);
+            ${devicename}->queue_command(set_payload_command);
+          }
   # Register: 0 -> Bit 1
   # When Water Flow Temperature Control is enabled for thise zone, the target outlet water temperature
   # will be used to define when the heatpump should be turned (powered) on or off
   #
   # If room thermostat is enabled, controlling this switch will be done by the heatpump. In other words,
   # the room thermostat will decide when this will be on or off.
-  - platform: modbus_controller
-    modbus_controller_id: "${devicename}"
-    name: "Water Flow Temperature Control Zone 1"
+  - platform: template
+    name: "Water Flow Temperature Control Zone 1"        #"Power Heat/Cool Control Zone 1"
     id: "${devicename}_water_flow_temperature_control_zone_1"
     icon: mdi:eye
-    register_type: holding
-    address: 0x0
-    bitmask: 0x2
-  # Register: 0 -> Bit 2
-  - platform: modbus_controller
-    modbus_controller_id: "${devicename}"
+    entity_category: config
+    restore_mode: DISABLED
+    optimistic : true
+    lambda: "return (id(unmasked_value_register_0) & 0x2) == 0x2; //return bit 0x2 status"
+    on_turn_on:
+      #- logger.log: "Switch Turned On!"
+      - lambda: |-
+          uint16_t checked_bit = 0x2;                          //affected bit number
```

Comment:
Nitpicking, can you align your comment style to always have a space after the // and start with a capital letter